packages/data-context/test/unit/sources/FileDataSource.spec.ts

```diff
@@ -118,6 +118,40 @@ describe('FileDataSource', () => {
           expect(files).to.have.length(2)
         })
 
+        it('does not ignores files within node_modules, if node_modules is in the glob path', async () => {
```

Comment:
Thanks for the additional integration test! Can you add some unit tests below as well, to flesh out coverage?

Reply:
Thank you for the review!
Could you please concretize what you would like to cover with tests? 

The introduced changes seem to be sufficiently covered with tests. I've introduced two checks: a) if node_modules is in the provided glob, b) if node_modules is in the cwd path. If a == true or b == true, node_modules is not added to the ignoreGlob.
It creates 4 combinations of checks:
a = false, b = false tested by the old integration test, which I renamed
a = true, b = false tested by the first added test
a = false, b = true tested by the second added test
a = true, b = true not tested, since a and b checked with OR condition testing this combination looks excessive to me

Reply:
They're definitely covered quite well, but as integration tests. The second part of this spec file includes unit tests, that mock out dependencies: the integration tests ensure that the changes work with the third party libraries, while the unit tests test that the changes work based on how we expect third party libraries to behave.

Reply:
Thanks for the explanation.

Unit tests are added.